server/server.go

```diff
@@ -0,0 +1,154 @@
+		// Stdout is closed, "there's no more to the show/
+		// If you all want to breath right/you all better go"
+		// This is going to seem a bit odd, but it is important to
+		// only wait for the process started here, not any orphans.
+		// In most cases, that process is either a singleton (so the wait
+		// will be all we need); a shell (which does all the waiting for
+		// its children); or the rare case of a detached process (in which
+		// case the reaper will get it).
+		// Seen in the wild: were this code to wait for orphans,
+		// and the main loop to wait for orphans, they end up
+		// competing with each other and the results are odd to say the least.
+		// If the command exits, leaving orphans behind, it is the job
+		// of the reaper to get them.
+		v("wait for %q", cmd)
+		err = cmd.Wait()
+		v("cmd %q returns with %v %v", cmd, err, cmd.ProcessState)
+		if errval(err) != nil {
+			v("CPUD:child exited with  %v", err)
+			s.Exit(cmd.ProcessState.ExitCode()) //nolint
+		}
+
+	} else {
+		cmd.Stdin, cmd.Stdout, cmd.Stderr = s, s, s
+		v("running command without pty")
+		if err := cmd.Run(); errval(err) != nil {
+			v("CPUD:err %v", err)
+			s.Exit(1) //nolint
+		}
+	}
+	verbose("handler exits")
+}
+
+// New sets up a cpud. cpud is really just an SSH server with a special
+// handler and support for port forwarding for the 9p port.
+func New(publicKeyFile, hostKeyFile string) (*ssh.Server, error) {
+	v("configure SSH server")
+	publicKeyOption := func(ctx ssh.Context, key ssh.PublicKey) bool {
+		data, err := ioutil.ReadFile(publicKeyFile)
+		if err != nil {
+			fmt.Print(err)
+			return false
+		}
+		allowed, _, _, _, err := ssh.ParseAuthorizedKey(data)
+		if err != nil {
+			fmt.Print(err)
+			return false
+		}
+		return ssh.KeysEqual(key, allowed)
+	}
+
+	// Now we run as an ssh server, and each time we get a connection,
+	// we run that command after setting things up for it.
+	forwardHandler := &ssh.ForwardedTCPHandler{}
+	server := &ssh.Server{
+		LocalPortForwardingCallback: ssh.LocalPortForwardingCallback(func(ctx ssh.Context, dhost string, dport uint32) bool {
+			log.Println("CPUD:Accepted forward", dhost, dport)
+			return true
+		}),
+		// Pick a reasonable default, which can be used for a call to listen and which
+		// will be overridden later from a listen.Addr
+		Addr:             ":23",
+		PublicKeyHandler: publicKeyOption,
+		ReversePortForwardingCallback: ssh.ReversePortForwardingCallback(func(ctx ssh.Context, host string, port uint32) bool {
+			v("CPUD:attempt to bind", host, port, "granted")
+			return true
+		}),
+		RequestHandlers: map[string]ssh.RequestHandler{
+			"tcpip-forward":        forwardHandler.HandleSSHRequest,
+			"cancel-tcpip-forward": forwardHandler.HandleSSHRequest,
+		},
+		Handler: handler,
+	}
+
+	if err := server.SetOption(ssh.HostKeyFile(hostKeyFile)); err != nil {
+		// We don't much care about this, what's the right thing to do here?
+		// just printint from this function is kind of icky.
```

Comment:
What's the issue with just printing a warning like `fmt.Printf("Could not load supplied host key: %s", err)`?

Reply:
random prints from random packages for warnings, at scale, make debugging in places like Google painful.